cobra/flux_analysis/room.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+
+from optlang.symbolics import Zero
+
+
+def add_room(model, solution=None, delta=0.03, epsilon=0.001, linear=False):
+    '''Add constraints and objective for ROOM
+
+    Parameters
+    ----------
+    model : cobra.Model
+        The model to add ROOM constraints and objectve to.
+    solution : cobra.Solution
+        A previous solution to use as a reference.
+    linear : bool
+        Whether to use the linear ROOM formulation or not.
+
+    Returns
+    -------
+    Nothing.
+
+    References
+    ----------
+    .. [1] Tomer Shlomi, Omer Berkman and Eytan Ruppin, "Regulatory on/off
+     minimization of metabolic flux changes after genetic perturbations",
+     PNAS 2005 102 (21) 7695-7700; doi:10.1073/pnas.0406346102
+
+    '''
+
+    if 'room_old_objective' in model.solver.variables:
+        raise ValueError('model is already adjusted for ROOM')
+
+    # optimizes if no reference solution is provided
+    if solution is None:
+        solution = model.optimize()
+
+    problem = model.problem
+    variable = problem.Variable("room_old_objective")
+    constraint = problem.Constraint(
```

Comment:
This does not yet constrain the old objective. For that you still need to set bounds on `variable` :)

Reply:
Okay! I will fix it.

Reply:
But, when I compare it to the cameo implementation, it's still slow by almost an order of 4. And, also what might be the reason that the LP variant is causing infeasibility in the model?

Reply:
Is that using symengine or sympy? Yeah there should not be an infeasibility for the linear version, is that with the fix for the growth rate constraint? Otherwise the problem might be the unbounded variable.

Reply:
I have set the upper bound of `variable` equal to `solution.objective_value` and it gave me the same result.
I used sympy. Even after making the variable bounded, the linear variant still gives infeasibility. I tried relaxing the growth rate constraint too but it doesn't help.